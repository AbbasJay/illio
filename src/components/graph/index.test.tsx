import React from "react";

import { mount } from "enzyme";
import renderer from "react-test-renderer";

import ShowGraph from "./";

const props = {
  chartData: [
    ["date", "close"],
    ["2020-01-01", 190],
    ["2020-01-02", 195],
    ["2020-01-03", 210],
  ],
};

const renderedComponent = () => {
  return mount(<ShowGraph {...props} />);
};

describe("ShowGraph", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly to snapshot", () => {
    const snapshot = renderer.create(<ShowGraph {...props} />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it("should render component", () => {
    const component = renderedComponent();

    expect(component.length).toBe(1);
  });

  describe("props", () => {
    it("should have the correct chartData prop", () => {
      const component = renderedComponent();

      expect(component.prop("chartData")).toEqual(props.chartData);
    });
  });
});
