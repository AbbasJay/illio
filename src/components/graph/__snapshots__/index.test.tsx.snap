// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowGraph renders correctly to snapshot 1`] = `
<div>
  Loading Chart
</div>
`;
